### ISSUE TITLE
Infinite loop mixing LongBigArrayBigList#size and LongBigArrayBigList#ensureCapacity

### ISSUE THREAD

**Issue by nathklei** [vigna/fastutil#144]
I encountered an infinite loop combining `LongBigArrayBigList#size` and `LongBigArrayBigList#ensureCapacity`, where the size of the LongBigArrayBigList was higher than the argument to ensureCapacity.

The repro is something like this:

```
        LongBigArrayBigList longList = new LongBigArrayBigList();
        longList.size(4096);
        // Never completes!
        longList.ensureCapacity(2);
```

I ended up changing my code to not ensureCapacity when less than size, but I spent a fair amount of time debugging why my program got stuck. At least exiting early would be better, or making the ensureCapacity a no-op.

Thanks!

**Comment by nathklei** [comment]
I traced the code down into `BigArrays#copy` - but I did not dig further.

**Comment by vigna** [comment]
Thanks a lot for the report. So, the problem is a bug in copy() that is unleashed when you pass crazy parameters, that is, you ask to preserve more elements than the target capacity. Of course it should work anyway (and I'll fix it), but the bug is actually unleashed by _another_ bug: ensureCapacity() tests length using a.length instead of length(a), which causes a call to forceCapacity() code with crazy parameters. Good catch!

**Comment by vigna** [comment]
That should be fixed in 4052352d5ab2f370b8acf72d4125a0bdcd4278f4 and 7ebec8bf70e173f0180ecaeaea7ad434a38025c7.

You can regenerate the sources and compile, or wait a few hours for an updated release (the bug is enough bad to yield a release).

**Comment by nathklei** [comment]
Great, appreciate the quick turnaround. I’ll update to the new version just in case!
